src/cli/telemetry/helpers.ts

```diff
@@ -0,0 +1,52 @@
+import fs from "fs";
+
+interface TerminalInfo {
+	/**
+	 * Whether this is in CI or not.
+	 */
+	readonly ci: boolean;
+	/**
+	 * Path to the user's shell program.
+	 */
+	readonly shell: string;
+	/**
+	 * Whether the terminal is an interactive TTY or not.
+	 */
+	readonly tty: boolean;
+	/**
+	 * Whether this is a Windows shell or not.
+	 */
+	readonly windows: boolean;
+}
+
+declare const TERMINAL_INFO: TerminalInfo;
+
+export const tryFn = async <T extends (...args: any[]) => Promise<R>, R>(
+	fn: T,
+	...args: any[]
+): Promise<R | null> => {
+	try {
+		return await fn(...args);
+	} catch {
+		// ignore
+	}
+
+	return null;
+};
+
+export const isInteractive = (): boolean => TERMINAL_INFO.tty && !TERMINAL_INFO.ci;
```

Comment:
One final ask - can we get this fn under test?

Reply:
Added!